### PR TITLE
[RESTEASY-1686] MediaType negotiation issues (2nd attempt)

### PR DESCRIPTION
**Pull request by asoldano**

### CODE REVIEW

File: resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ServerResponseWriter.java [hunk 1]
```diff
@@ -178,51 +179,61 @@ protected static void setDefaultContentType(HttpRequest request, BuiltResponse j
...
-                        {
-                           break;
-                        }
-                     }
-                  }
+                  produces = method.getProduces();
                }
                else 
                {
                   method.getMethod().getClass().getAnnotation(Produces.class);
```
Comment (NicoNes) [inline resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ServerResponseWriter.java]: Unused

Comment (asoldano) [inline resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ServerResponseWriter.java]: Fixed, thanks

File: resteasy-jaxrs/src/main/java/org/jboss/resteasy/spi/WeldUtil.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    * This is needed because of https://issues.jboss.org/browse/WELD-1539 and
+    * https://issues.jboss.org/browse/WELD-1914 which are unsolved.
+    *
+    * @param clazz the class of interest
+    *
+    * @return {@code true} if the given class is a Weld proxy,
+    * {@code false} otherwise
+    */
+   static boolean isWeldProxy(Class<?> clazz) {
+      for ( Class<?> implementedInterface : clazz.getInterfaces() ) {
```
Comment (mkouba) [inline resteasy-jaxrs/src/main/java/org/jboss/resteasy/spi/WeldUtil.java]: You could also easily filter out classes which are not synthetic, i.e. those for which `Class.isSynthetic()` returns false  - see also https://issues.jboss.org/browse/WELD-1914?focusedCommentId=13099225&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13099225.

Comment (asoldano) [inline resteasy-jaxrs/src/main/java/org/jboss/resteasy/spi/WeldUtil.java]: Thanks @mkouba , see https://github.com/resteasy/Resteasy/pull/1220

**Pull request merged**